src/v/security/audit/schemas/tests/ocsf_schemas_test.cc

```diff
@@ -0,0 +1,577 @@
+    BOOST_REQUIRE_EQUAL(ser, minify(expected));
+}
+
+SEASTAR_THREAD_TEST_CASE(validate_increment) {
+    auto now = sa::timestamp_t{
+      std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::system_clock::now().time_since_epoch())
+        .count()};
+    auto app_lifecycle = sa::application_lifecycle(
+      sa::application_lifecycle::activity_id::start,
+      sa::product{test_product},
+      sa::severity_id::informational,
+      now);
+
+    using namespace std::chrono_literals;
+    ss::sleep(5ms).get();
+
+    auto increment_time = sa::timestamp_t{
+      std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::system_clock::now().time_since_epoch())
+        .count()};
+
+    app_lifecycle.increment(increment_time);
+
+    ss::sleep(5ms).get();
+
+    auto increment_time2 = sa::timestamp_t{
+      std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::system_clock::now().time_since_epoch())
+        .count()};
+
+    app_lifecycle.increment(increment_time2);
+
+    auto ser = sa::rjson_serialize(app_lifecycle);
+
+    const ss::sstring expected{
+      R"(
+{
+  "category_uid": 6,
+  "class_uid": 6002,
+  "count": 3,
+  "end_time": )"
+      + ss::to_sstring(increment_time2) + R"(,
+  "metadata": )"
+      + metadata_ser + R"(,
+  "severity_id": 1,
+  "start_time": )"
+      + ss::to_sstring(now) + R"(,
+  "time": )"
+      + ss::to_sstring(now) + R"(,
+  "type_uid": 600203,
+  "activity_id": 3,
+  "app": )"
+      + test_product_ser + R"(
+}
+)"};
+
+    BOOST_REQUIRE_EQUAL(ser, minify(expected));
+}
+
+SEASTAR_THREAD_TEST_CASE(validate_api_activity_hash) {
+    using namespace std::chrono_literals;
+    {
+        size_t hash1 = 0, hash2 = 0;
+        {
+            fragmented_vector<sa::authorization_result> authz;
+            authz.push_back(authz_success);
+            fragmented_vector<sa::resource_detail> resources;
+            resources.push_back(resource_detail);
+            sa::network_endpoint dst_endpoint;
+            auto now = sa::timestamp_t{
+              std::chrono::duration_cast<std::chrono::milliseconds>(
+                std::chrono::system_clock::now().time_since_epoch())
+                .count()};
+            auto api_act = sa::api_activity{
+              sa::api_activity::activity_id::create,
+              sa::actor{
+                .authorizations = std::move(authz), .user = default_user},
+              sa::api{api_create_topic},
+              make_copy_of_network_endpoint(rp_kafka_endpoint),
+              test_http_request(),
+              std::move(resources),
+              sa::severity_id::informational,
+              make_copy_of_network_endpoint(client_kafka_endpoint),
+              sa::api_activity::status_id::success,
+              now,
+              sa::api_activity_unmapped{unmapped}};
+            hash1 = api_act.hash();
+            BOOST_REQUIRE_EQUAL(hash1, api_act.key());
+        }
+        ss::sleep(5ms).get();
```

Comment:
What is this sleep for?

Reply:
Eh I can refactor this to hard code in times rather than use `std::chrono::system_clock::now()`

Reply:
Funny story - i did remove it but forgot to save the file :)